x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/LookupFromIndexService.java

```diff
@@ -115,25 +138,24 @@ protected AbstractLookupService.LookupResponse readLookupResponse(StreamInput in
     }
 
     public static class Request extends AbstractLookupService.Request {
-        private final String matchField;
+        private final List<LookupFromIndexOperator.MatchConfig> matchFields;
 
         Request(
             String sessionId,
             String index,
             String indexPattern,
-            DataType inputDataType,
-            String matchField,
+            List<LookupFromIndexOperator.MatchConfig> matchFields,
             Page inputPage,
             List<NamedExpression> extractFields,
             Source source
         ) {
-            super(sessionId, index, indexPattern, inputDataType, inputPage, extractFields, source);
-            this.matchField = matchField;
+            super(sessionId, index, indexPattern, matchFields.get(0).type(), inputPage, extractFields, source);
```

Comment:
Here we populate the `inputDataType` even though this information is redundant and incomplete, whereas the `matchFields` has all this info now.